From: Yuval Mintz <yuvalmin@broadcom.com>
Date: Tue, 3 Apr 2012 18:41:25 +0000
Subject: [PATCH 070/111] bnx2x: enable inta on the pci bus when used
Patch-mainline: v3.5-rc1
Git-commit: 79a8557a6d18c3861d64ae110ddd7606c65d7504
References: bnc#790597 FATE#313915

During boot-from-SAN, if msix interrupts are unavailable and inta
is requested, it is possible that inta would be disabled in the
pci bus. This patch enables inta when requested.

Signed-off-by: Yuval Mintz <yuvalmin@broadcom.com>
Signed-off-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -1368,6 +1368,9 @@ static void bnx2x_igu_int_enable(struct
 
 	REG_WR(bp, IGU_REG_PF_CONFIGURATION, val);
 
+	if (val & IGU_PF_CONF_INT_LINE_EN)
+		pci_intx(bp->pdev, true);
+
 	barrier();
 
 	/* init leading/trailing edge */
